### ISSUE TITLE
Windows Build error with --cofig=mkl => ImportError: DLL load failed while importing _pywrap_tensorflow_internal

### ISSUE THREAD

**Issue by bani-intelaipg** [tensorflow/tensorflow#55817]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

tf2.9

### Custom Code

No

### OS Platform and Distribution

Windows Server 2019

### Python version

3.10

### Bazel version

5.2.0

### GCC/Compiler version

Windows VC++ 2019

### Current Behaviour?

```shell
Trying to build from TF master branch with --config=mkl flag, and we see this error.
The build was made within a clean Py-3.10 virtual-env, with ONLY these pip modules:
python -m pip install absl-py astunparse flatbuffers google_pasta h5py keras-nightly keras_preprocessing numpy opt_einsum protobuf scipy six termcolor typing_extensions wheel wrapt gast tensorboard tf-estimator-nightly packaging portpicker 

(tried with commit# 1cca89708c5b218e38cd68d341c98c2d06f3870b)

\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\pywrap_tensorflow.py", line 62, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.
```

### Standalone code to reproduce the issue

```shell
CI script is invoked in following dir C:/.../tensorflow/ using Windows command prompt:
bash C:/Users/mlp_admin/tensorflow/tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh --extra_build_flags '--action_env=TEMP=C:\Users\mlp_admin\tmp --action_env=TMP=C:\Users\mlp_admin\tmp --config=mkl' --extra_test_flags '--action_env=TEMP=C:\Users\mlp_admin\tmp --action_env=TMP=C:\Users\mlp_admin\tmp --config=mkl' --skip_test

This internally calls =>
bazel build --experimental_cc_shared_library --config=release_cpu_windows '--action_env=TEMP=C:\Users\mlp_admin\tmp' '--action_env=TMP=C:\Users\mlp_admin\tmp' --config=mkl '--output_filter=^$' tensorflow/tools/pip_package:build_pip_package
```

### Relevant log output

```shell
ERROR: C:/users/mlp_admin/tensorflow/tensorflow/BUILD:1352:19: Executing genrule //tensorflow:tf_python_api_gen_v2 failed: (Exit 1): bash.exe failed: error executing command 
  cd /d C:/users/mlp_admin/tensorflow-build/m2fy67yv/execroot/org_tensorflow
  SET PATH=C:\msys64\usr\bin;C:\msys64\bin;C:\Windows;C:\Windows\System32;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Users\mlp_admin\AppData\Local\bazelisk\downloads\bazelbuild\bazel-4fd7983ca82fab536b8f9bb589610036b01d7490-windows-x86_64\bin;C:\tools\cuda\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.2\extras\CUPTI\libx64;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.2\bin;C:\C;C:\msys64\Users\mlp_admin\venv_py310\Scripts;C:\Program Files\Git\cmd;C:\C;C:\msys64\Users\mlp_admin\venv_py310;C:\tools\bazel\;C;C:\Users\mlp_admin\venv_py310\Scripts;C:\Tools\bazel;C:\Program Files\Git;C:\Program Files\Git\cmd;C:\msys64;C:\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
    SET PYTHON_BIN_PATH=C:/Users/mlp_admin/venv_py310/Scripts/python.exe
    SET PYTHON_LIB_PATH=C:/Users/mlp_admin/venv_py310/Lib/site-packages
    SET RUNFILES_MANIFEST_ONLY=1
    SET TEMP=C:\Users\mlp_admin\tmp
    SET TF2_BEHAVIOR=1
    SET TMP=C:\Users\mlp_admin\tmp
  C:\msys64\usr\bin\bash.exe bazel-out/x64_windows-opt/bin/tensorflow/tf_python_api_gen_v2.genrule_script.sh
# Configuration: 92001cead5cfe54bd325acb14a4c0622c2eacc2d756dddb96cef1f45acd66008
# Execution platform: @local_execution_config_platform//:platform
Traceback (most recent call last):
  File "\\?\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\pywrap_tensorflow.py", line 62, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "\\?\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\tools\api\generator\create_python_api.py", line 22, in <module>
    from tensorflow.python.tools.api.generator import doc_srcs
  File "\\?\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\__init__.py", line 36, in <module>
    from tensorflow.python import pywrap_tensorflow as _pywrap_tensorflow
  File "\\?\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\pywrap_tensorflow.py", line 77, in <module>
    raise ImportError(
ImportError: Traceback (most recent call last):
  File "\\?\C:\Users\mlp_admin\tmp\Bazel.runfiles_x184n_p0\runfiles\org_tensorflow\tensorflow\python\pywrap_tensorflow.py", line 62, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.


Failed to load the native TensorFlow runtime.
See https://www.tensorflow.org/install/errors for some common causes and solutions.
If you need help, create an issue at https://github.com/tensorflow/tensorflow/issues and include the entire stack trace above this error message.
Target //tensorflow/tools/pip_package:build_pip_package failed to build
ERROR: C:/users/mlp_admin/tensorflow/tensorflow/lite/python/BUILD:68:10 PythonZipper tensorflow/lite/python/tflite_convert.zip failed: (Exit 1): bash.exe failed: error executing command 
  cd /d C:/users/mlp_admin/tensorflow-build/m2fy67yv/execroot/org_tensorflow
  SET PATH=C:\msys64\usr\bin;C:\msys64\bin;C:\Windows;C:\Windows\System32;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Users\mlp_admin\AppData\Local\bazelisk\downloads\bazelbuild\bazel-4fd7983ca82fab536b8f9bb589610036b01d7490-windows-x86_64\bin;C:\tools\cuda\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.2\extras\CUPTI\libx64;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.2\bin;C:\C;C:\msys64\Users\mlp_admin\venv_py310\Scripts;C:\Program Files\Git\cmd;C:\C;C:\msys64\Users\mlp_admin\venv_py310;C:\tools\bazel\;C;C:\Users\mlp_admin\venv_py310\Scripts;C:\Tools\bazel;C:\Program Files\Git;C:\Program Files\Git\cmd;C:\msys64;C:\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
    SET PYTHON_BIN_PATH=C:/Users/mlp_admin/venv_py310/Scripts/python.exe
    SET PYTHON_LIB_PATH=C:/Users/mlp_admin/venv_py310/Lib/site-packages
    SET RUNFILES_MANIFEST_ONLY=1
    SET TEMP=C:\Users\mlp_admin\tmp
    SET TF2_BEHAVIOR=1
    SET TMP=C:\Users\mlp_admin\tmp
  C:\msys64\usr\bin\bash.exe bazel-out/x64_windows-opt/bin/tensorflow/tf_python_api_gen_v2.genrule_script.sh
# Configuration: 92001cead5cfe54bd325acb14a4c0622c2eacc2d756dddb96cef1f45acd66008
# Execution platform: @local_execution_config_platform//:platform
INFO: Elapsed time: 4138.656s, Critical Path: 866.88s
INFO: 11869 processes: 306 internal, 11563 local.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```
</details>

**Comment by bani-intelaipg** [comment]
Tagging @mihaimaruseac

**Comment by mihaimaruseac** [comment]
Rostam, this looks like another cc_shared_library issue

**Comment by mihaimaruseac** [comment]
We were using a non-LTS Bazel at that commit, to work around some other `cc_shared_library` issues. Can you try with 3637a0245a2b44f2de2aa84b6f8f40cac5600109 (the commit after the one you tested, which reverts back to LTS Bazel)? This way we'd know if the issue is Bazel or some `cc_shared_library` changelist.

**Comment by bani-intelaipg** [comment]
Same error is there with commit# b5fa6a4eecdcf69408708e98ba4de6debb880596

**Comment by learning-to-play** [comment]
Thanks for letting us know. We're working on the issue. Will update.

**Comment by mihaimaruseac** [comment]
Penporn (@penpornk), can you bisect this if this still happens? I couldn't find a cc_shared_library last week last week that could have cause this

**Comment by penpornk** [comment]
@bani-intelaipg Is this a recent failure or pre-existing one? When was the last buildable commit for you?
I looked at the [TF-oneDNN nightly CI](https://tensorflow-ci.intel.com/job/test-tensorflow-mkl-build-win-whl-nightly/) and saw a similar error message since [3/10/22](https://tensorflow-ci.intel.com/job/test-tensorflow-mkl-build-win-whl-nightly/8/). 
```
Traceback (most recent call last):
01:16:43    File "\\?\C:\Users\mlp_admin\AppData\Local\Temp\Bazel.runfiles__j3gzb3n\runfiles\org_tensorflow\tensorflow\python\pywrap_tensorflow.py", line 62, in <module>
01:16:43      from tensorflow.python._pywrap_tensorflow_internal import *
01:16:43  ImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.
```

**Comment by bani-intelaipg** [comment]
@penpornk I believe this has been broken for a while. I tried to go back to find a passing build with "--config=mkl" setting, but haven't found one with my limited rebuild effort. I can use some guidance on a potential analysis of the why this import is potentially failing in Windows. Do we have to explicitly set some PATH/PYTHONPATH env var to help the build find the module? The pipeline you mentioned was a recent temporary effort and not a mainline pipeline.

**Comment by mihaimaruseac** [comment]
Oh, I somehow got the impression that the failure was recent. If it was an older one then the theory that this is caused by `cc_shared_library` becomes viable again

**Comment by learning-to-play** [comment]
@bani-intelaipg Debugging shared library failures can be tricky.
- Do you see the error when NOT building with `--config=mkl`?
- If you can find a commit that builds and one that doesn't, it will help narrow down the underlying cause. Can you try building a green commit from 1 week ago, 2 weeks ago, and so on until you find one that builds? Then we can use [git-bisect](https://git-scm.com/docs/git-bisect) to narrow down which change caused this.

**Comment by masaton6** [comment]
I got hit by exact same issue while building with `--config=mkl` . I tried ProcessMonitor downloaded from Microsoft site to identify which DLL is missing and got below error.

`8:44:18.8425549	python.exe	14196	CreateFile	C:\Python\Python310\libomp.dll	NAME NOT FOUND	Desired Access: Read Attributes, Disposition: Open, Options: Open For Backup, Open Reparse Point, Attributes: n/a, ShareMode: Read, Write, Delete, AllocationSize: n/a`

`libomp.dll` is in the PATH but seems not loaded using PATH so I put it into `C:\Python\Python310` (python.exe installed directory) as work around, then build went through.
Not sure this apply for everybody, and not sure how to resolve this from root, though hope this info help to resolve this issue.
